decompile/preprocessing/preprocess.py

```diff
@@ -2,45 +2,70 @@
 import os
 import subprocess
 import shutil
+from multiprocessing import Pool
 
-C_CODE_FOLDER = "D:\Work\Research paper\DatasetC"
-ASSEMBLY_FOLDER = "D:\Work\Research paper\DatasetAO3"
-SOURCE_FOLDER = "D:\Work\Research paper\AnghaBench-master"
+folder_path = "/home/mohanned/Work/DataC"
+output_folder = "/home/mohanned/Work/Binary"
 
+def compile_to_binary(file_path: str) -> None:
+    """
+        Converting C code files into binary files
+        
+        Args:
+        file_path (str): It's the path for the C code file being compiled
+    """
+    binary_file = os.path.splitext(file_path)[0]
+    binary_file = binary_file[::-1]
+    EDIT = ""
+    for c in binary_file:
+        if(c == '/'):
+            break
+        EDIT += c
+    EDIT = EDIT[::-1]
+    out_file = output_folder + "/" + EDIT + ".out"
 
-def compile_to_assembly(c_code_file: str, output_file: str) -> bool:
-    """Compile C code to assembly.
+    assemble_command = f"gcc -c {file_path} -o {out_file}"
 
-    Args:
-        c_code_file_path (str): Path to C code file
-        output_file_path (str): Path to C code file
+    try:
+        subprocess.run(assemble_command, check=True, shell=True)
+        print(f"Assembly successful. Binary machine code saved as {out_file}")
+    except subprocess.CalledProcessError as e:
+        print(f"Assembly failed with error:\n{e}")
 
-    Returns:
-        bool: Indicates whether the compilation was successful
+def disassemble_to_assembly(binary_file: str) -> None:
     """
-    try:
-        subprocess.run(["gcc", "-S", "-O3", c_code_file, "-o", output_file], check=True)
-        return True
-    except subprocess.CalledProcessError:
-        return False
-
-
-def compile_folder() -> None:
-    """Compile C code"""
-    if not os.path.exists(ASSEMBLY_FOLDER):
-        os.makedirs(ASSEMBLY_FOLDER)
-    for root, _, files in os.walk(C_CODE_FOLDER):
-        for file in files:
-            if not file.endswith(".c"):
-                continue
+        Disassembling binary files
+        
+        Args:
+        file_path (str): It's the path for the binary file being disassembled
+    """
+    assembly_file = os.path.splitext(binary_file)[0] + ".s"
+    with open(assembly_file, "w") as assembly_file:
+        subprocess.run(["objdump", "-d", "-M", "att", "-M", "x86-64", "--no-addresses", binary_file], stdout=assembly_file)

```

Comment:
Include "att" and "x86-64" as a parameter instead of a magic value. They can be changed in the future.